services/ui_backend_service/data/db/tables/artifact.py

```diff
@@ -15,3 +16,48 @@ class AsyncArtifactTablePostgres(AsyncPostgresTable):
     trigger_keys = MetadataArtifactTable.trigger_keys
     select_columns = keys
     _command = MetadataArtifactTable._command
+
+    async def get_artifact_locations_for_run_ids(self, run_ids=[]):
+        # do not touch DB if no run_ids were given.
+        if len(run_ids) == 0:
```

Comment:
small flub in original code: the function actually deals with run_numbers so the param name should reflect this. run_ids will not work for the fetching. rename run_ids to run_numbers :)